src/Blocks/BlockPiston.cpp

```diff
@@ -160,6 +160,12 @@ bool cBlockPistonHandler::CanPushBlock(
 		return !a_RequirePushable;
 	}
 
+	if ((a_BlockPos.y >= cChunkDef::Height - 1) || (a_BlockPos.y <= 0))
```

Comment:
```suggestion
	if (!cChunkDef::IsValidHeight(a_BlockPos.y + a_PushDir.y))
```

Reply:
Also, can we handle pushing multiple blocks?

Reply:
Thanks, I missed that function. Yes, that case is handled.